src/Sulu/Bundle/AdminBundle/Resources/js/containers/FieldBlocks/FieldBlocks.js

```diff
@@ -13,6 +13,40 @@ const MISSING_BLOCK_ERROR_MESSAGE = 'The "block" field type needs at least one t
 const BLOCK_PREVIEW_TAG = 'sulu.block_preview';
 
 export default class FieldBlocks extends React.Component<FieldTypeProps<Array<BlockEntry>>> {
+    componentDidUpdate(prevProps: FieldTypeProps<Array<BlockEntry>>) {
+        const {defaultType, types, value: value, onChange} = this.props;
+        const {types: oldTypes} = prevProps;
+
+        if (!types || !oldTypes) {
+            throw new Error(MISSING_BLOCK_ERROR_MESSAGE);
+        }
+
+        const newValue = toJS(value);
+        let hasValueChanged = false;
+
+        if (value && types !== oldTypes) {
+            if (!defaultType) {
+                throw new Error(
+                    'It is impossible that a block has no defaultType. This should not happen and is likely a bug.'
+                );
+            }
+
+            // set block to default type if type does not longer exist
+            // this could happen for example in a template switch
+            newValue.forEach((block, i) => {
+                if (!types[block.type]) {
+                    hasValueChanged = true;
+                    newValue[i].type = defaultType;
+                }
+            });
+        }
+
+        // onChange should only be called when value was changed else it will end in a infinite loop
```

Comment:
Why isn't that done in the `if` block above? Then it would only be called if the block types changed, and this entire logic does not have to be executed on every update.

Actually I would say it is a best practice to first check if props (or whatever you are listening to) as soon as possible. E.g. also the [react documentation](https://reactjs.org/docs/react-component.html#componentdidupdate) makes a check as soon as possible.

Reply:
as above think `onChange` should only be called once in `componentDidUpdate` so if somebody later add something here to change the value he should insert that code before this call. If I do it directly in if I think if somebody adds here something `onChange` is maybe called twice instead of once.